ci/build-mingw64.sh

```diff
@@ -62,135 +75,184 @@ function makeplusinstall () {
     [ -d libplacebo ] || $gitclone https://code.videolan.org/videolan/libplacebo.git
     builddir libplacebo
-    meson .. --cross-file "$prefix_dir/crossfile"
+    meson setup .. --cross-file "$prefix_dir/crossfile" \
+        -Ddemos=false -D{opengl,d3d11}=enabled
     makeplusinstall
     popd
-fi
+}
+_libplacebo_mark=lib/libplacebo.dll.a
 
-## freetype2
-if [ ! -e "$prefix_dir/lib/libfreetype.dll.a" ]; then
-    ver=2.13.0
+_freetype () {
+    local ver=2.13.1
     gettar "https://mirror.netcologne.de/savannah/freetype/freetype-${ver}.tar.xz"
     builddir freetype-${ver}
-    meson .. --cross-file "$prefix_dir/crossfile"
+    meson setup .. --cross-file "$prefix_dir/crossfile"
     makeplusinstall
     popd
-fi
+}
+_freetype_mark=lib/libfreetype.dll.a
 
-## fribidi
-if [ ! -e "$prefix_dir/lib/libfribidi.dll.a" ]; then
-    ver=1.0.13
+_fribidi () {
+    local ver=1.0.13
     gettar "https://github.com/fribidi/fribidi/releases/download/v${ver}/fribidi-${ver}.tar.xz"
     builddir fribidi-${ver}
-    meson .. --cross-file "$prefix_dir/crossfile" \
+    meson setup .. --cross-file "$prefix_dir/crossfile" \
         -D{tests,docs}=false
     makeplusinstall
     popd
-fi
+}
+_fribidi_mark=lib/libfribidi.dll.a
 
-## harfbuzz
-if [ ! -e "$prefix_dir/lib/libharfbuzz.dll.a" ]; then
-    ver=7.3.0
+_harfbuzz () {
+    local ver=8.1.1
     gettar "https://github.com/harfbuzz/harfbuzz/releases/download/${ver}/harfbuzz-${ver}.tar.xz"
     builddir harfbuzz-${ver}
-    meson .. --cross-file "$prefix_dir/crossfile" \
+    meson setup .. --cross-file "$prefix_dir/crossfile" \
         -Dtests=disabled
     makeplusinstall
     popd
-fi
+}
+_harfbuzz_mark=lib/libharfbuzz.dll.a
 
-## libass
-if [ ! -e "$prefix_dir/lib/libass.dll.a" ]; then
+_libass () {
     [ -d libass ] || $gitclone https://github.com/libass/libass.git
     builddir libass
     [ -f ../configure ] || (cd .. && ./autogen.sh)
     ../configure --host=$TARGET $commonflags
     makeplusinstall
     popd
-fi
+}
+_libass_mark=lib/libass.dll.a
 
-## luajit
-if [ ! -e "$prefix_dir/lib/libluajit-5.1.a" ]; then
-    $gitclone https://github.com/LuaJIT/LuaJIT.git
+_luajit () {
+    [ -d LuaJIT ] || $gitclone https://github.com/LuaJIT/LuaJIT.git
     pushd LuaJIT
-    hostcc="ccache cc"
-    flags=
+    local hostcc="ccache cc"
+    local flags=
     [[ "$TARGET" == "i686-"* ]] && { hostcc="$hostcc -m32"; flags=XCFLAGS=-DLUAJIT_NO_UNWIND; }
     make TARGET_SYS=Windows clean
     make TARGET_SYS=Windows HOST_CC="$hostcc" CROSS="ccache $TARGET-" \
         BUILDMODE=static $flags amalg
     make DESTDIR="$prefix_dir" INSTALL_DEP= FILE_T=luajit.exe install
     popd
+}
+_luajit_mark=lib/libluajit-5.1.a
+
+for x in iconv zlib ffmpeg shaderc spirv-cross; do
+    build_if_missing $x
+done
+if [[ "$TARGET" != "i686-"* ]]; then
+    build_if_missing vulkan-headers
+    build_if_missing vulkan-loader
 fi
+for x in libplacebo freetype fribidi harfbuzz libass luajit; do
+    build_if_missing $x
```

Comment:
To simplify I would remove `build_if_missing` logic and build always. We had the checks in script, but on CI it doesn't change anything and if someone uses this script locally, they can modify it to their needs. I don't mind having it, but it makes script more complex.

Reply:
I like to iteratively work on the script locally and find this function very useful. There's no harm in keeping it in.